From 6dbe4f734ec02def0faba15a7db0b5068bc551fa Mon Sep 17 00:00:00 2001
From: Yuxi Sun <b36102@freescale.com>
Date: Thu, 15 Dec 2011 14:28:16 +0800
Subject: [PATCH 248/505] ENGR00170361 v4l2 capture: fix capture segment fault
 error

After user space buffer method is introduced in 2.6.38, capture
segment error happened. So need to set buffer memory flag before
query buffer.

Signed-off-by: Yuxi Sun <b36102@freescale.com>
---
 test/mxc_v4l2_test/mxc_v4l2_capture.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/test/mxc_v4l2_test/mxc_v4l2_capture.c b/test/mxc_v4l2_test/mxc_v4l2_capture.c
index 3b0906a..50cdecc 100644
--- a/test/mxc_v4l2_test/mxc_v4l2_capture.c
+++ b/test/mxc_v4l2_test/mxc_v4l2_capture.c
@@ -83,7 +83,8 @@ int start_capturing(int fd_v4l)
         {
                 memset(&buf, 0, sizeof (buf));
                 buf.type = V4L2_BUF_TYPE_VIDEO_CAPTURE;
-                buf.index = i;
+                buf.memory = V4L2_MEMORY_MMAP;
+		buf.index = i;
                 if (ioctl(fd_v4l, VIDIOC_QUERYBUF, &buf) < 0)
                 {
                         printf("VIDIOC_QUERYBUF error\n");
-- 
1.8.0

